management-api-server/src/main/java/com/datastax/mgmtapi/resources/models/Compaction.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Please see the included license file for details.
+ */
+package com.datastax.mgmtapi.resources.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Describes the state of an active compaction running on the server.
+ *
+ * <p>Some fields are specific to certain Cassandra versions, this is indicated in their comment.
+ */
+public class Compaction {
+
+  // Note: for simplicity, we use the same keys in our JSON payload as the map returned by
+  // Cassandra. These constants are used for both, do not change them or the corresponding JSON
+  // fields will always be empty.
+  private static final String ID_KEY = "id";
+  private static final String KEYSPACE_KEY = "keyspace";
+  private static final String COLUMN_FAMILY_KEY = "columnfamily";
+  private static final String COMPLETED_KEY = "completed";
+  private static final String TOTAL_KEY = "total";
+  private static final String TASK_TYPE_KEY = "taskType";
+  private static final String UNIT_KEY = "unit";
+  private static final String COMPACTION_ID_KEY = "compactionId";
+  private static final String SSTABLES_KEY = "sstables";
+  private static final String TARGET_DIRECTORY_KEY = "targetDirectory";
+  private static final String OPERATION_TYPE_KEY = "operationType";
+  private static final String OPERATION_ID_KEY = "operationId";
+  private static final String DESCRIPTION_KEY = "description";
```

Comment:
Thought:
This is fine. This approach is better than introducing Cassandra/DSE dependencies here as that would force us to break the server/API side of the code up into version specific projects like we do for the Agents.